smart-table-module/js/Table.js

```diff
@@ -113,6 +113,15 @@
             };
 
             /**
+             * set lastColumnSort to defaultSortColumn if any based on column's reverse value
+             * @method setDefaultSortColumn
+             * @param column
+             */
+            function setDefaultSortColumn (defaultSortColumn) {
+                lastColumnSort = (lastColumnSort !== '') ? defaultSortColumn : lastColumnSort;
```

Comment:
@2013gang thanks for pointing out, now I have validated the `lastColumnSort` object using `angular.isObject`, this would also return false for *null* object.